resonance/tests/test_nonlinear_systems.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Circle
+
+from ..nonlinear_systems import SimplePendulum
+
+sys = SimplePendulum()
+
+sys.constants['mass'] = 1.0  # kg
+sys.constants['length'] = 1.0  # m
+sys.constants['acc_due_to_grav'] = 9.81  # m/s**2
+
+sys.coordinates['angle'] = np.deg2rad(15.0)  # rad
+sys.speeds['angle_vel'] = 0.0  # rad/s
+
+
+def equations_of_motion(state, time, length, acc_due_to_grav):
+
+    l = length
+    g = acc_due_to_grav
+
+    theta = state[0]
+    omega = state[1]
+
+    thetad = omega
+    omegad = -g / l * np.sin(theta)
+
+    return [thetad, omegad]
+
+# function that computes the right hand side of the equations of motion in
+# first order form
+sys.equations_of_motion = equations_of_motion
+
+
+def height(angle, length):
+    """The Y coordinate points in the opposite of gravity, i.e. up. The X
+    coordinate points to the right."""
+    return -length * np.cos(angle)
+
+sys.add_measurement('height', height)
+
+
+def sway(angle, length):
+    """The X coordinate points to the right."""
+    return length * np.sin(angle)
+
+sys.add_measurement('sway', sway)
+
+
+def potential_energy(length, height, mass, acc_due_to_grav):
+    return mass * acc_due_to_grav * (length + height)
+
+sys.add_measurement('potential_energy', potential_energy)
+
+
+def kinetic_energy(mass, length, angle_vel):
+    return mass * (length * angle_vel)**2 / 2.0
+
+sys.add_measurement('kinetic_energy', kinetic_energy)
+
+
+def total_energy(kinetic_energy, potential_energy):
+    return kinetic_energy + potential_energy
+
+sys.add_measurement('total_energy', total_energy)
```

Comment:
I'm getting a very weird plot here. The values are correct for this but matplotlib seems to plot the floating point error or something instead of the values themselves. confused.
![plot_same_values](https://user-images.githubusercontent.com/276007/31280847-96cf6548-aa62-11e7-88e5-9bc322341d70.png)

Reply:
@ixjlyons I forgot to ask you this yesterday, but do you understand why mpl does this?

Reply:
I've figured it out. It is autoscaling the y limits and actually subtracts the man value before plotting. The notation "1e-7+3.34267e-1" was confusing me, but it means to append that to the y tick values and then do the addition.